File: src/main/java/social/alone/server/user/controller/UserEventController.java
package social.alone.server.user.controller;

import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import social.alone.server.event.Event;
import social.alone.server.event.EventQueryParams;
import social.alone.server.event.EventSearchService;
import social.alone.server.user.User;

import javax.validation.Valid;
import java.util.Optional;

@Controller
@RequestMapping(value = "/api/users/{userId}/events")
@RequiredArgsConstructor
public class UserEventController {

  final private EventSearchService eventSearchService;

  @GetMapping()
  public ResponseEntity getEventsWithUser(
          Pageable pageable,
          @PathVariable("userId") User user,
          @Valid final EventQueryParams eventQueryParams
  ) {
    Page<Event> page = eventSearchService.findAllBy(pageable, Optional.ofNullable(user), eventQueryParams);
    return ResponseEntity.ok(page);
  }

}

File: src/main/kotlin/social/alone/server/user/controller/UserEventController.kt
package social.alone.server.user.controller

import org.springframework.data.domain.Pageable
import org.springframework.data.domain.Sort
import org.springframework.data.web.PageableDefault
import org.springframework.http.ResponseEntity
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import social.alone.server.event.service.EventSearchService
import social.alone.server.event.type.EventQueryParams
import social.alone.server.user.domain.User
import javax.validation.Valid

@Controller
@RequestMapping(value = ["/api/users/{userId}/events"])
class UserEventController(val eventSearchService: EventSearchService) {

    @GetMapping
    fun getEventsWithUser(
            @PageableDefault(sort = ["startedAt"], direction = Sort.Direction.ASC) pageable: Pageable,
            @PathVariable("userId") user: User,
            @Valid eventQueryParams: EventQueryParams
    ): ResponseEntity<*> {
        val page = eventSearchService.findAllBy(pageable, eventQueryParams)
        return ResponseEntity.ok(page)
    }

}
